aws.go

```diff
@@ -29,41 +70,39 @@ func getLatestDatapoint(datapoints []*cloudwatch.Datapoint) *cloudwatch.Datapoin
 	return latest
 }
 
-// scrape makes the required calls to AWS CloudWatch by using the parameters in the Collector
+// Scrape makes the required calls to AWS CloudWatch by using the parameters in the Collector.
 // Once converted into Prometheus format, the metrics are pushed on the ch channel.
-func scrape(instance, region string, collector *Collector, ch chan<- prometheus.Metric) {
-	session := session.Must(session.NewSession(&aws.Config{
-		Region: aws.String(region),
-	}))
-
-	svc := cloudwatch.New(session)
-
-	labels := []string{}
-	labels = append(labels, instance, region)
-
+func (s *Scrape) Scrape() {
 	wg := &sync.WaitGroup{}
-	wg.Add(len(collector.Metrics))
-	for _, metric := range collector.Metrics {
+	wg.Add(len(s.collector.Metrics))
+	for _, metric := range s.collector.Metrics {
 		go func(metric Metric) {
-			err := scrapeMetric(svc, metric, instance, collector, ch, labels)
+			err := s.scrapeMetric(metric)
 			if err != nil {
 				fmt.Println(err)
 			}
+
 			wg.Done()
 		}(metric)
 	}
 	wg.Wait()
```

Comment:
I think we should limit the number of parallel scrapes, and set idle connection parameters in `http.Transport`. Otherwise, a number of TCP connections will jump up and down all the time, that will affect the performance of the whole PMM Server box.